backend/monoprocess/monoprocess_test.go

```diff
@@ -0,0 +1,100 @@
+package monoprocess
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"reflect"
+	"testing"
+	"time"
+	"unsafe"
+
+	"github.com/cschleiden/go-workflows/backend"
+	"github.com/cschleiden/go-workflows/backend/sqlite"
+	"github.com/cschleiden/go-workflows/backend/test"
+	"github.com/cschleiden/go-workflows/internal/history"
+)
+
+func Test_MonoprocessBackend(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	test.BackendTest(t, func(options ...backend.BackendOption) test.TestBackend {
+		// Disable sticky workflow behavior for the test execution
+		options = append(options, backend.WithStickyTimeout(0))
+
+		return NewMonoprocessBackend(sqlite.NewInMemoryBackend(options...), 0, time.Millisecond)
+	}, nil)
+}
+
+func Test_EndToEndMonoprocessBackend(t *testing.T) {
+	if testing.Short() {
+		t.Skip()
+	}
+
+	test.EndToEndBackendTest(t, func(options ...backend.BackendOption) test.TestBackend {
+		// Disable sticky workflow behavior for the test execution
+		options = append(options, backend.WithStickyTimeout(0))
+
+		return NewMonoprocessBackend(sqlite.NewInMemoryBackend(options...), 0, 0)
+	}, nil)
+}
+
+var _ test.TestBackend = (*monoprocessBackend)(nil)
+
+func (b *monoprocessBackend) GetFutureEvents(ctx context.Context) ([]*history.Event, error) {
+	// Reflection hack to access db in private field of sqlite.sqliteBackend
+	privateDbField := reflect.ValueOf(b.Backend).Elem().FieldByName("db")
+	hackedDbField := reflect.NewAt(privateDbField.Type(), unsafe.Pointer(privateDbField.UnsafeAddr())).Elem()
+	db := hackedDbField.Interface().(*sql.DB)
```

Comment:
Sure, I can change the sqlite backend to expose this method. The drawback is that it will be exposed for everyone to use, I assume you wanted to avoid that so I didn't touch it at first.